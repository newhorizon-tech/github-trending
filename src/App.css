:root {
  /* Mint and Sailor Blue */
  --light: #adefd1ff;
  --dark: #00203fff;

  /* Black and White */

  /* --light:  #ececec;
   --dark: #00070d; */
}

* {
  margin: 0;
  padding: 0;
}

html,
body {
  width: 100%;
  height: 100%;
  background-color: var(--light);
  color: var(--dark);
}

#header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  height: 130px;
  background: var(--dark);
  color: var(--light);
}

#search-bar {
  height: 20px;
  border: none;
  padding: 2px;
}

#project-cards {
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.project-card {
  font-size: 1.1rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border: 1px solid var(--dark);
  color: var(--dark);
  height: 150px;
}

.card-1,
.card-0 {
  background-color: var(--light);
  color: var(--dark);
}

.card-2,
.card-3 {
  background-color: var(--dark);
  color: var(--light);
}

a:nth-child(an+b) .project-body {
  display: flex;
  flex-direction: column;
}

.project-body p,
.built-box {
  padding: 10px 0;
  border: 1px solid var(--dark);
}

.creators {
  display: flex;
  align-items: center;
  justify-content: center;
}

.creators img {
  width: 60px;
  margin: 0 2px;
}

.back-button {
  background: var(--light);
  padding: 3px;
  color: var(--dark);
  border: none;
}

.back-symbol {
  display: inline-block;
  transform: rotate(180deg);
}

.loading {
  display: flex;
  justify-content: center;
  margin: 50px;
}

a.no-format-link { text-decoration: none; color: inherit;}
