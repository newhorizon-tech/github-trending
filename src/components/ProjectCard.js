import { useNavigate } from 'react-router-dom';

import PropTypes from 'prop-types';

const ProjectCard = (props) => {
  const navigate = useNavigate();
  const { project } = props;
  return (
    <button
      className={`project-card card-${project.rank % 4}`}
      type="button"
      onClick={() => navigate(`/projects/${project.rank}`)}
    >
      <span className="name">
        {project.repositoryName}
      </span>
      <span>
        {project.starsSince}
      </span>
    </button>

  );
};

ProjectCard.propTypes = {
  project: PropTypes.shape({
    rank: PropTypes.number,
    repositoryName: PropTypes.string,
    starsSince: PropTypes.number,
  }).isRequired,
};

export default ProjectCard;
