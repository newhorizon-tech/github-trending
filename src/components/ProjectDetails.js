import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';

const ProjectDetails = (props) => {
  const { project } = props;
  const creators = [...project.builtBy];
  const navigate = useNavigate();
  const timePeriod = {
    daily: 'today',
    weekly: 'this week',
    monthly: 'this month',
  };
  const trimString = (str) => {
    const limit = 23;
    let trimmedString = str;
    if (str.length > limit) {
      trimmedString = `${str.substring(0, limit - 3)}...`.split('-').join(' ');
    }
    return trimmedString;
  };

  return (
    <>
      <div id="header">
        <button
          className="back-button"
          type="button"
          onClick={() => navigate('/')}
        >
          <span className="back-symbol"> &#10140;</span>
          Back
        </button>
        <h1>
          {trimString(project.repositoryName)}
        </h1>
        <h2>
          #
          {`${project.rank} ${timePeriod[project.since]}`}
        </h2>

      </div>
      <div className="project-body">
        <p>
          Repo owner:
          {` ${project.username}`}
        </p>
        <p>
          Stars:
          {` ${project.totalStars}`}
        </p>
        <p>
          Forks:
          {` ${project.forks}`}
        </p>
        <p>
          Description:
          {` ${project.description}`}
        </p>
        <p>
          Language:
          {` ${project.language}`}
        </p>
        <div className="built-box">
          <span> Built by: </span>
          <div className="creators">
            {creators.map((user) => (
              <div key={user.username} className="built-user">
                <a href={user.url}>
                  <img
                    className="avatars"
                    src={user.avatar}
                    alt={user.username}
                    title={user.username}
                  />
                </a>
              </div>
            ))}
          </div>
        </div>
      </div>
    </>
  );
};

ProjectDetails.propTypes = {
  project: PropTypes.shape({
    description: PropTypes.string,
    language: PropTypes.string,
    rank: PropTypes.number,
    since: PropTypes.string,
    username: PropTypes.string,
    totalStars: PropTypes.number,
    forks: PropTypes.number,
    repositoryName: PropTypes.string,
    starsSince: PropTypes.number,
    builtBy: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)),
  }).isRequired,
};

export default ProjectDetails;
