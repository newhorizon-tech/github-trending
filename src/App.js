import './App.css';

import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { SpinnerInfinity } from 'spinners-react';
import { trendingAsync, search } from './redux/trendingSlice';

import ProjectCard from './components/ProjectCard';

const App = () => {
  const dispatch = useDispatch();
  const searchTerm = useSelector((state) => state.trending.search);
  const projects = useSelector((state) => {
    const { projects } = state.trending;
    if (searchTerm !== '' && !!projects) {
      return projects.filter((x) => {
        const languageMatch = !(x.language) ? false : x.language.toLowerCase().includes(searchTerm);
        const nameMatch = !(x.repositoryName) ? false
          : x.repositoryName.toLowerCase().includes(searchTerm);
        return languageMatch || nameMatch;
      });
    }
    return projects;
  });
  const loaded = useSelector((state) => state.trending.status === 'loaded');
  const loading = useSelector((state) => state.trending.status === 'loading');

  useEffect(() => {
    if (!loaded) {
      dispatch(trendingAsync());
    }
  }, [loaded]);

  const handleSearch = (e) => {
    dispatch(search(e.target.value.toLowerCase()));
  };

  return (
    <div>
      <div id="header">
        <h1> Github Trending </h1>
        <input id="search-bar" placeholder="Search" onChange={(e) => handleSearch(e)} value={searchTerm} />
      </div>
      {loading
        ? (
          <span className="loading">
            <SpinnerInfinity
              size={71}
              thickness={135}
              speed={135}
              color={getComputedStyle(document.documentElement).getPropertyValue('--light')}
              secondaryColor={getComputedStyle(document.documentElement).getPropertyValue('--dark')}
            />
          </span>
        )
        : ''}
      <div id="project-cards">
        {projects.map((project) => <ProjectCard key={project.rank} project={project} />)}
      </div>
    </div>
  );
};

export default App;
