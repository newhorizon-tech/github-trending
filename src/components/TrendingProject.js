import { useParams } from 'react-router-dom';
import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { SpinnerInfinity } from 'spinners-react';
import { trendingAsync } from '../redux/trendingSlice';
import ProjectDetails from './ProjectDetails';

const TrendingProject = () => {
  const params = useParams();
  const projectId = parseInt(params.projectId, 10);
  const dispatch = useDispatch();
  const loading = useSelector((state) => state.trending.status === 'loading');
  const loaded = useSelector((state) => state.trending.status === 'loaded');
  const project = useSelector((state) => state.trending.projects.find((x) => x.rank === projectId));

  useEffect(() => {
    if (!loaded) {
      dispatch(trendingAsync());
    }
  }, [loaded]);

  return (
    <div>
      {loading
        ? (
          <span className="loading">
            <SpinnerInfinity
              size={71}
              thickness={135}
              speed={135}
              color={getComputedStyle(document.documentElement).getPropertyValue('--light')}
              secondaryColor={getComputedStyle(document.documentElement).getPropertyValue('--dark')}
            />
          </span>
        ) : ''}
      {loaded ? <ProjectDetails project={project} /> : ''}
    </div>
  );
};

export default TrendingProject;
