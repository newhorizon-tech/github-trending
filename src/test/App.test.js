import { render } from '@testing-library/react';
import App from '../App';

import mockAPI from './mockAPI';
import mockIndex from './mockIndex';

import ProjectCard from '../components/ProjectCard';
import ProjectDetails from '../components/ProjectDetails';

import TrendingProject from '../components/TrendingProject';

import reducer, { search, addProjects } from '../redux/trendingSlice';

global.fetch = jest.fn(mockAPI);
const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('renders App snapshot', () => {
  const tree = render(mockIndex(App, []));
  expect(tree).toMatchSnapshot();
});

it('renders ProjectDetails snapshot', () => {
  const tree = render(mockIndex(ProjectDetails, []));
  expect(tree).toMatchSnapshot();
});

it('renders ProjectCard snapshot', () => {
  const tree = render(mockIndex(ProjectDetails, []));
  expect(tree).toMatchSnapshot();
});

test('renders trending project card properly', async () => {
  const response = await mockAPI('correct-url');
  const projects = await response.json();
  const project = projects[0];
  const rendered = render(mockIndex(ProjectCard, project));

  const heading = rendered.container.querySelector('h1');

  expect(heading.textContent).toBe(' Github Trending ');
});

test('renders trending project page', async () => {
  const response = await mockAPI('correct-url');
  const projects = await response.json();
  const project = projects[0];
  const rendered = render(mockIndex(TrendingProject, project));

  const heading = rendered.container.querySelector('h1');

  expect(heading.textContent).toBe(' Github Trending ');
});

test('returns correct project response', async () => {
  const response = await mockAPI('correct-url');
  const projects = await response.json();
  const project = projects[0];

  expect(project.repositoryName).toBe('incubator-seatunnel');
});

test('returns correct project response', async () => {
  const response = await mockAPI('correct-url');
  const projects = await response.json();
  const project = projects[0];

  expect(project.repositoryName).toBe('incubator-seatunnel');
});

test('reducer returns correct newState when you search somethign', async () => {
  const initialState = { search: '', status: '', projects: [] };
  const newState = reducer(initialState, search('SearchString'));

  expect(newState).toStrictEqual({ projects: [], search: 'SearchString', status: '' });
});

test('reducer returns correct newState when you add projects', async () => {
  const response = await mockAPI('correct-url');
  const projects = await response.json();
  const initialState = { search: '', status: '', projects: [] };
  const newState = reducer(initialState, addProjects(projects));

  expect(newState).toStrictEqual({ projects, search: '', status: '' });
});
