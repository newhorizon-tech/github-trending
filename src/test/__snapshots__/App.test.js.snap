// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          id="header"
        >
          <h1>
             Github Trending 
          </h1>
          <input
            id="search-bar"
            placeholder="Search"
            value=""
          />
        </div>
        <span
          class="loading"
        >
          <svg
            fill="none"
            style="overflow: visible; width: 71px;"
            viewBox="0 0 131 55"
          >
            <defs>
              <path
                d="M46.57 45.5138C36.346 55.4954 19.8919 55.4954 9.66794 45.5138C-0.55598 35.5321 -0.55598 19.4678 9.66794 9.48624C19.8919 -0.495412 36.346 -0.495412 46.57 9.48624L84.4303 45.5138C94.6543 55.4954 111.108 55.4954 121.332 45.5138C131.556 35.5321 131.556 19.4678 121.332 9.48624C111.108 -0.495412 94.6543 -0.495412 84.4303 9.48624L46.57 45.5138Z"
                id="spinners-react-infinity-path"
              />
            </defs>
            <use
              stroke=""
              stroke-width="9.450000000000001"
              xlink:href="#spinners-react-infinity-path"
            />
            <use
              fill="none"
              stroke="currentColor"
              stroke-dasharray="1, 347"
              stroke-dashoffset="75"
              stroke-linecap="round"
              stroke-width="9.450000000000001"
              style="animation: spinners-react-infinity 1.037037037037037s linear infinite;"
              xlink:href="#spinners-react-infinity-path"
            />
          </svg>
        </span>
        <div
          id="project-cards"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        id="header"
      >
        <h1>
           Github Trending 
        </h1>
        <input
          id="search-bar"
          placeholder="Search"
          value=""
        />
      </div>
      <span
        class="loading"
      >
        <svg
          fill="none"
          style="overflow: visible; width: 71px;"
          viewBox="0 0 131 55"
        >
          <defs>
            <path
              d="M46.57 45.5138C36.346 55.4954 19.8919 55.4954 9.66794 45.5138C-0.55598 35.5321 -0.55598 19.4678 9.66794 9.48624C19.8919 -0.495412 36.346 -0.495412 46.57 9.48624L84.4303 45.5138C94.6543 55.4954 111.108 55.4954 121.332 45.5138C131.556 35.5321 131.556 19.4678 121.332 9.48624C111.108 -0.495412 94.6543 -0.495412 84.4303 9.48624L46.57 45.5138Z"
              id="spinners-react-infinity-path"
            />
          </defs>
          <use
            stroke=""
            stroke-width="9.450000000000001"
            xlink:href="#spinners-react-infinity-path"
          />
          <use
            fill="none"
            stroke="currentColor"
            stroke-dasharray="1, 347"
            stroke-dashoffset="75"
            stroke-linecap="round"
            stroke-width="9.450000000000001"
            style="animation: spinners-react-infinity 1.037037037037037s linear infinite;"
            xlink:href="#spinners-react-infinity-path"
          />
        </svg>
      </span>
      <div
        id="project-cards"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders ProjectCard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          id="header"
        >
          <h1>
             Github Trending 
          </h1>
          <input
            id="search-bar"
            placeholder="Search"
            value=""
          />
        </div>
        <span
          class="loading"
        >
          <svg
            fill="none"
            style="overflow: visible; width: 71px;"
            viewBox="0 0 131 55"
          >
            <defs>
              <path
                d="M46.57 45.5138C36.346 55.4954 19.8919 55.4954 9.66794 45.5138C-0.55598 35.5321 -0.55598 19.4678 9.66794 9.48624C19.8919 -0.495412 36.346 -0.495412 46.57 9.48624L84.4303 45.5138C94.6543 55.4954 111.108 55.4954 121.332 45.5138C131.556 35.5321 131.556 19.4678 121.332 9.48624C111.108 -0.495412 94.6543 -0.495412 84.4303 9.48624L46.57 45.5138Z"
                id="spinners-react-infinity-path"
              />
            </defs>
            <use
              stroke=""
              stroke-width="9.450000000000001"
              xlink:href="#spinners-react-infinity-path"
            />
            <use
              fill="none"
              stroke="currentColor"
              stroke-dasharray="1, 347"
              stroke-dashoffset="75"
              stroke-linecap="round"
              stroke-width="9.450000000000001"
              style="animation: spinners-react-infinity 1.037037037037037s linear infinite;"
              xlink:href="#spinners-react-infinity-path"
            />
          </svg>
        </span>
        <div
          id="project-cards"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        id="header"
      >
        <h1>
           Github Trending 
        </h1>
        <input
          id="search-bar"
          placeholder="Search"
          value=""
        />
      </div>
      <span
        class="loading"
      >
        <svg
          fill="none"
          style="overflow: visible; width: 71px;"
          viewBox="0 0 131 55"
        >
          <defs>
            <path
              d="M46.57 45.5138C36.346 55.4954 19.8919 55.4954 9.66794 45.5138C-0.55598 35.5321 -0.55598 19.4678 9.66794 9.48624C19.8919 -0.495412 36.346 -0.495412 46.57 9.48624L84.4303 45.5138C94.6543 55.4954 111.108 55.4954 121.332 45.5138C131.556 35.5321 131.556 19.4678 121.332 9.48624C111.108 -0.495412 94.6543 -0.495412 84.4303 9.48624L46.57 45.5138Z"
              id="spinners-react-infinity-path"
            />
          </defs>
          <use
            stroke=""
            stroke-width="9.450000000000001"
            xlink:href="#spinners-react-infinity-path"
          />
          <use
            fill="none"
            stroke="currentColor"
            stroke-dasharray="1, 347"
            stroke-dashoffset="75"
            stroke-linecap="round"
            stroke-width="9.450000000000001"
            style="animation: spinners-react-infinity 1.037037037037037s linear infinite;"
            xlink:href="#spinners-react-infinity-path"
          />
        </svg>
      </span>
      <div
        id="project-cards"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders ProjectDetails snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          id="header"
        >
          <h1>
             Github Trending 
          </h1>
          <input
            id="search-bar"
            placeholder="Search"
            value=""
          />
        </div>
        <span
          class="loading"
        >
          <svg
            fill="none"
            style="overflow: visible; width: 71px;"
            viewBox="0 0 131 55"
          >
            <defs>
              <path
                d="M46.57 45.5138C36.346 55.4954 19.8919 55.4954 9.66794 45.5138C-0.55598 35.5321 -0.55598 19.4678 9.66794 9.48624C19.8919 -0.495412 36.346 -0.495412 46.57 9.48624L84.4303 45.5138C94.6543 55.4954 111.108 55.4954 121.332 45.5138C131.556 35.5321 131.556 19.4678 121.332 9.48624C111.108 -0.495412 94.6543 -0.495412 84.4303 9.48624L46.57 45.5138Z"
                id="spinners-react-infinity-path"
              />
            </defs>
            <use
              stroke=""
              stroke-width="9.450000000000001"
              xlink:href="#spinners-react-infinity-path"
            />
            <use
              fill="none"
              stroke="currentColor"
              stroke-dasharray="1, 347"
              stroke-dashoffset="75"
              stroke-linecap="round"
              stroke-width="9.450000000000001"
              style="animation: spinners-react-infinity 1.037037037037037s linear infinite;"
              xlink:href="#spinners-react-infinity-path"
            />
          </svg>
        </span>
        <div
          id="project-cards"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        id="header"
      >
        <h1>
           Github Trending 
        </h1>
        <input
          id="search-bar"
          placeholder="Search"
          value=""
        />
      </div>
      <span
        class="loading"
      >
        <svg
          fill="none"
          style="overflow: visible; width: 71px;"
          viewBox="0 0 131 55"
        >
          <defs>
            <path
              d="M46.57 45.5138C36.346 55.4954 19.8919 55.4954 9.66794 45.5138C-0.55598 35.5321 -0.55598 19.4678 9.66794 9.48624C19.8919 -0.495412 36.346 -0.495412 46.57 9.48624L84.4303 45.5138C94.6543 55.4954 111.108 55.4954 121.332 45.5138C131.556 35.5321 131.556 19.4678 121.332 9.48624C111.108 -0.495412 94.6543 -0.495412 84.4303 9.48624L46.57 45.5138Z"
              id="spinners-react-infinity-path"
            />
          </defs>
          <use
            stroke=""
            stroke-width="9.450000000000001"
            xlink:href="#spinners-react-infinity-path"
          />
          <use
            fill="none"
            stroke="currentColor"
            stroke-dasharray="1, 347"
            stroke-dashoffset="75"
            stroke-linecap="round"
            stroke-width="9.450000000000001"
            style="animation: spinners-react-infinity 1.037037037037037s linear infinite;"
            xlink:href="#spinners-react-infinity-path"
          />
        </svg>
      </span>
      <div
        id="project-cards"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
